cgroup2/manager.go

```diff
@@ -552,28 +552,63 @@ func (c *Manager) MoveTo(destination *Manager) error {
 func (c *Manager) Stat() (*stats.Metrics, error) {
 	var metrics stats.Metrics
 
-	cpuStat := make(map[string]uint64)
-	if err := readKVStatsFile(c.path, "cpu.stat", cpuStat); err != nil {
+	var err error
+	metrics.CPU, err = readCPUStats(c.path)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics.Memory, err = readMemoryStats(c.path)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics.MemoryEvents, err = readMemoryEvents(c.path)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics.Io = &stats.IOStat{
+		Usage: readIoStats(c.path),
+		PSI:   getStatPSIFromFile(filepath.Join(c.path, "io.pressure")),
+	}
+	metrics.Rdma = &stats.RdmaStat{
+		Current: rdmaStats(filepath.Join(c.path, "rdma.current")),
+		Limit:   rdmaStats(filepath.Join(c.path, "rdma.max")),
+	}
+	metrics.Hugetlb = readHugeTlbStats(c.path)
+
+	return &metrics, nil
+}
+
+func readMemoryEvents(cgroupPath string) (*stats.MemoryEvents, error) {
```

Comment:
Alternatively, this could be a method (instead of passing the path as argument);

```suggestion
func (c *Manager) readMemoryEvents() (*stats.MemoryEvents, error) {
```

My consideration was that having it as a plain function could be easier to add unit tests for, but of course we could probably construct a `Manager{path: "some custom path"}`, so probably either way would work